bytestring.cabal

```diff
@@ -131,26 +141,32 @@ library
                     -fmax-simplifier-iterations=10
                     -fdicts-cheap
                     -fspec-constr-count=6
-
-  c-sources:        cbits/fpstring.c
-                    cbits/itoa.c
-                    cbits/shortbytestring.c
-                    cbits/aligned-static-hs-data.c
-
-  if (arch(aarch64))
-    c-sources:        cbits/aarch64/is-valid-utf8.c
+  
+  if arch(javascript) || flag(pure-haskell)
```

Comment:
Perhaps also add `default-extensions: NoForeignFunctionInterface` just to be extra explicit.